snooty/taxonomy.py

```diff
@@ -44,9 +44,10 @@ def get_taxonomy(cls) -> "TaxonomySpec":
     @classmethod
     def validate_key_value_pairs(cls, facet_str_pairs: List[Tuple[str, str]]) -> None:
         taxonomy_ref = asdict(cls.get_taxonomy())
+        facet_pairs = copy.deepcopy(facet_str_pairs)
```

Comment:
Because tuples and strings are both immutable, you don't actually have to do a deepcopy! Just do `facet_pairs = facet_str_pairs[:]`

Reply:
Also, perhaps it would be better for this method to take a `Sequence[]` than a `List[]` if we're not mutating it?

Reply:
I do have to do a deep copy because `facet_str_pairs` is a `List` and we are calling `pop()`. This lead to a side effect where subsequent calls to this method would be missing additional values needed for validation.